pyigm/cgm/utils.py

```diff
@@ -95,5 +101,122 @@ def cgm_from_galaxy_igmsystems(galaxy, igmsystems, R_max=300*u.kpc, dv_max=400*u
     # Return
     return cgm_list
 
+def velcorr_mould(galaxy,cosmo=None):
+    '''
+    Calculates angular diameter distance, luminosity distance, etc., corrected
+    for peculiar motions due to the Shapley Supercluster, the Local Group, etc.,
+    using the formalism of Mould et al. (2000..ApJ..529..786).
+
+    Code ported from IDL implementation by John Moustakas:
+    github.com/moustakas/impro/blob/master/pro/galaxy/mould_distance.pro
 
+    Parameters
+    ----------
+    galaxy: Galaxy
+    cosmo: astropy.cosmology, optional
+
+    Returns
+    -------
+    velcorrdict: dictionary
+        Resulting quantities of velocity corrected calculation.
+        v_hel: Heliocentric velocity of object with no correction, simply z*c
+        v_LG: Velocity with respect to Local Group
+        v_virgo: Velocity with respect to Virgo Cluster
+        v_GA: Velocity with respect to Virgo Cluster
+        v_shapley: Velocity with respect to Shapley Supercluster
+        v_cosmic: Corrected velocity
+        lumdist: Luminosity distance
+        angdiamdist: Angular diameter distance
+        distmod: Distance modulus
+        flag: 1 if object is the direction of Virgo, 2 if GA, 3 if Shapley
+        Scale: Proper distance per angular separation on sky
+    '''
+
+    # Cosmology
+    if cosmo is None:
+        cosmo = cosmology.Planck15
+
+    H0 = cosmo.H0
+    omega0 = cosmo.Om0
+    omega_lambda = cosmo.Ode0
+
+    # Needed parameters
+    q0 = omega0 / 2.0 - omega_lambda
+    gamma = 2.0
+
+    # Load info from Mould+ 2000 Table 1A
+    clnames = ['Virgo', 'GA', 'Shapley']
+    clra1950 =  ['12h28m19s','13h20m00s','13h30m00s']
+    cldec1950 = ['+12:40:00','-44:00:00','-31:00:00']
+    clcoords = SkyCoord(clra1950,cldec1950,unit=(u.hourangle,u.degree),frame='fk4',equinox='B1950.0')
+    clhelvel = np.array([1035., 4600., 13800.]) * u.km/u.s
+    clLGvel = np.array([957., 4380., 13600.])  * u.km/u.s
+    fidvel = np.array([200., 400., 85.]) * u.km/u.s
+    clrad = np.array([10., 10., 12.]) * u.degree
+    clrangelo = np.array([600., 2600., 10000.]) * u.km/u.s
+    clrangehi = np.array([2300., 6600., 16000.]) * u.km/u.s
+
+    # Convert B1950 coordinates (as given) and
+    clcoords = clcoords.icrs
+
+   # Convert input coords to Galactic coords
+    galcoords_gal = galaxy.coord.transform_to(frame='galactic')
+
+    # Transform to local group frame
+    c = 2.99792458e5 * u.km/u.s
+    v_LG = c * galaxy.z - 79.0 * (u.km/u.s) * np.cos(galcoords_gal.l).value \
+        * np.cos(galcoords_gal.b).value + 296.0 * (u.km/u.s) * np.sin(galcoords_gal.l).value \
```

Comment:
Perhaps this is a bit awkward style.  I've moved the .value inside the function argument.